lua/lspconfig/nls.lua

```diff
@@ -0,0 +1,60 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+configs.nls = {
+  default_config = {
+    cmd = { 'nls' },
+    filetypes = { 'ncl', 'nickel' },
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or util.path.dirname(fname)
+    end,
+  },
+
+  docs = {
+    description = [[
+Nickel Language Server
+
+https://github.com/tweag/nickel
+
+`nls` can be installed with nix, or cargo from the Nickel repository.
+```sh
+git clone https://github.com/tweag/nickel.git
+```
+
+Nix:
+```sh
+cd nickel
+nix-env -f . -i
+```
+
+cargo:
+```sh
+cd nickel/lsp/nls
+cargo install --path .
+```
+
+Install the Nickel vim plugin https://github.com/nickel-lang/vim-nickel.
+
+**Snippet to enable the language server:**
+```lua
+require'lspconfig'.nls.setup{}
+```
+
+**Commands and default values:**
+```lua
+  Commands:
+
+  Default Values:
+    cmd = { "nls" }
+    filetypes = { "ncl", "nickel" }
+    root_dir = function(fname)
+          return util.find_git_ancestor(fname) or util.path.dirname(fname)
+        end,
+```
+        ]],
```

Comment:
Everything in the documentation past this line is (including this line) is auto-generated